app/views/govuk_component/lead_paragraph.raw.html.erb

```diff
@@ -0,0 +1,12 @@
+<%
+  text ||= ""
+  def nbsp_between_last_two_words(text)
+    escaped_text = html_escape_once(text.strip)
+    escaped_text.sub(/\s([\w\.\?\!\:]+)$/, '&nbsp;\1').html_safe
+  end
+%>
```

Comment:
I'm not sure we need the method definition in here as the logic isn't going to be reused. Could rewrite as:

```erb
<%
  text ||= ""
  escaped_text = html_escape_once(text.strip)
  text = escaped_text.sub(/\s([\w\.\?\!\:]+)$/, '&nbsp;\1').html_safe
%>
```

Then further down use:
`<%= text %>`

The tests can then test different component input, checking that the outputted HTML is as expected.

I'm not sure what scope the `nbsp_between_last_two_words` method has in a view, and I'm surprised the tests worked without needing to call render on a component.